### ISSUE TITLE
testsuite fails on ruby3.3.0dev due to Logger change

### ISSUE THREAD

**Issue by mtasaka** [asciidoctor/asciidoctor#4493]
With upcomming ruby 3.3.0dev (I tried: https://github.com/ruby/ruby/commit/52f8b347a0cc0d0529d5ddaf3094631e8fd914f2 ) asciidoctor git head (  https://github.com/asciidoctor/asciidoctor/commit/880f043f4543da2d57c9834190192fff2ec97572 ) testsuite fails with a lot of messages like:

```
  1) Error:
Minitest::Test::TestLinks#test_xref_using_angled_bracket_syntax_with_nested_path_which_has_been_included_in_this_document:
ArgumentError: invalid log level: 
    /usr/share/ruby/logger/severity.rb:34:in `coerce'
    /usr/share/ruby/logger.rb:401:in `level='
    /builddir/build/GIT/asciidoctor/test/test_helper.rb:331:in `ensure in in_verbose_mode'
    /builddir/build/GIT/asciidoctor/test/test_helper.rb:331:in `in_verbose_mode'
    /builddir/build/GIT/asciidoctor/test/links_test.rb:699:in `block (3 levels) in <top (required)>'
    /builddir/build/GIT/asciidoctor/test/test_helper.rb:320:in `using_memory_logger'
    /builddir/build/GIT/asciidoctor/test/links_test.rb:698:in `block (2 levels) in <top (required)>'

  2) Error:
Minitest::Test::TestLinks#test_should_not_interpret_path_sans_extension_in_xref_with_angled_bracket_syntax_in_compat_mode:
NoMethodError: undefined method `[]' for nil
    /usr/share/ruby/logger.rb:385:in `level'
    /usr/share/ruby/logger.rb:494:in `info?'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/substitutors.rb:809:in `block in sub_macros'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/substitutors.rb:729:in `gsub'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/substitutors.rb:729:in `sub_macros'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/substitutors.rb:103:in `block in apply_subs'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/substitutors.rb:92:in `each'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/substitutors.rb:92:in `apply_subs'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/block.rb:113:in `content'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/converter/html5.rb:798:in `convert_paragraph'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/converter/html5.rb:54:in `convert'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/abstract_block.rb:77:in `convert'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/abstract_block.rb:86:in `block in content'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/abstract_block.rb:86:in `map'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/abstract_block.rb:86:in `content'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/document.rb:1012:in `content'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/converter/html5.rb:326:in `convert_embedded'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/converter/html5.rb:64:in `convert'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/document.rb:956:in `convert'
    /builddir/build/GIT/asciidoctor/test/links_test.rb:540:in `block (3 levels) in <top (required)>'
    /builddir/build/GIT/asciidoctor/test/test_helper.rb:320:in `using_memory_logger'
    /builddir/build/GIT/asciidoctor/test/links_test.rb:538:in `block (2 levels) in <top (required)>'

  3) Error:
Minitest::Test::TestLinks#test_should_warn_and_create_link_if_debug_mode_is_enabled_inter_document_xref_points_to_current_doc_and_reference_not_found:
ArgumentError: invalid log level: 
    /usr/share/ruby/logger/severity.rb:34:in `coerce'
    /usr/share/ruby/logger.rb:401:in `level='
    /builddir/build/GIT/asciidoctor/test/test_helper.rb:331:in `ensure in in_verbose_mode'
    /builddir/build/GIT/asciidoctor/test/test_helper.rb:331:in `in_verbose_mode'
    /builddir/build/GIT/asciidoctor/test/links_test.rb:1006:in `block (3 levels) in <top (required)>'
    /builddir/build/GIT/asciidoctor/test/test_helper.rb:320:in `using_memory_logger'
    /builddir/build/GIT/asciidoctor/test/links_test.rb:1005:in `block (2 levels) in <top (required)>'

  4) Error:
Minitest::Test::TestLinks#test_xref_using_macro_syntax_with_path_and_extension_in_compat_mode:
NoMethodError: undefined method `[]' for nil
    /usr/share/ruby/logger.rb:385:in `level'
    /usr/share/ruby/logger.rb:494:in `info?'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/substitutors.rb:809:in `block in sub_macros'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/substitutors.rb:729:in `gsub'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/substitutors.rb:729:in `sub_macros'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/substitutors.rb:103:in `block in apply_subs'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/substitutors.rb:92:in `each'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/substitutors.rb:92:in `apply_subs'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/block.rb:113:in `content'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/converter/html5.rb:798:in `convert_paragraph'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/converter/html5.rb:54:in `convert'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/abstract_block.rb:77:in `convert'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/abstract_block.rb:86:in `block in content'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/abstract_block.rb:86:in `map'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/abstract_block.rb:86:in `content'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/document.rb:1012:in `content'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/converter/html5.rb:326:in `convert_embedded'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/converter/html5.rb:64:in `convert'
    /builddir/build/GIT/asciidoctor/lib/asciidoctor/document.rb:956:in `convert'
    /builddir/build/GIT/asciidoctor/test/links_test.rb:653:in `block (3 levels) in <top (required)>'
    /builddir/build/GIT/asciidoctor/test/test_helper.rb:320:in `using_memory_logger'
    /builddir/build/GIT/asciidoctor/test/links_test.rb:651:in `block (2 levels) in <top (required)>'

...
...
```

This is because of ruby3.3 Logger change:
https://github.com/ruby/ruby/commit/194520f80e1cdb71faa055d731450855a1ddb8d1
https://github.com/ruby/logger/pull/85

This can be fixed by calling initializer of `::Logger` (by super).

**Comment by mojavelinux** [comment]
This same change needs to be made for the NullLogger and MemoryLogger, but we're missing tests for it. I'll open another issue to address it.

**Comment by mojavelinux** [comment]
Oops, I tricked myself. This has already been fixed, but Asciidoctor 2.0.21 has not yet been released. It's time for me to schedule a release then.
